action/README.md

```diff
@@ -1,86 +1,134 @@
 
-on: push
+### Permissions
 
+You have to set `id-token: write` permission to authenticate with Deno Deploy.
+
+```yaml
 jobs:
   deploy:
-    runs-on: ubuntu-latest
-
     permissions:
-      id-token: write # This is required to allow the GitHub Action to authenticate with Deno Deploy.
+      id-token: write # required
       contents: read
-
     steps:
-      - name: Clone repository
-        uses: actions/checkout@v3
-
-      - name: Deploy to Deno Deploy
-        uses: denoland/deployctl@v1
-        with:
-          project: my-project # the name of the project on Deno Deploy
-          entrypoint: main.ts # the entrypoint to deploy
+      # your steps here...
+```
+
+### Inputs
+
+```yaml
+- name: Deploy to Deno Deploy
+  uses: denoland/deployctl@v1
+  with:
+    # Name of the project on Deno Deploy
+    # Required.
+    project:
+
+    # Entrypoint location executed by Deno Deploy
+    # The entrypoint can be a relative path or an absolute URL.
+    # If it is a relative path, it will resolved relative to the `root` directory.
+    # Required.
+    entrypoint:
+
+    # Root directory to deploy
+    # All files in this directory will be deployed.
+    # Optional. Default is "process.cwd()"
+    root:
+
+    # Filter which files to include in the deployment
+    # Optional.
+    include:
+
+    # Filter which files to exclude in the deployment
+    # Optional.
+    exclude:
+
+    # Location of an import map
+    # Must be relative to root directory
+    # Optional.
+    import-map:
 ```
 
-By default the entire contents of the repository will be deployed. This can be
-changed by specifying the `root` option.
+## Examples
 
-```yml
+### Deploy everything
+
+```yaml
 - name: Deploy to Deno Deploy
   uses: denoland/deployctl@v1
   with:
     project: my-project
-    entrypoint: index.js
-    root: dist
+    entrypoint: main.ts
 ```
 
-The `entrypoint` can either be a relative path or file name, or a an absolute
-URL. If it is a relative path, it will be resolved relative to the `root`. Both
-absolute `file:///` and `https://` URLs are supported.
+### Deploy a directory
 
-To deploy the `./dist` directory using the [std/http/file_server.ts][fileserver]
-module, you can use the following configuration:
+```yaml
+- name: Deploy to Deno Deploy
+  uses: denoland/deployctl@v1
+  with:
+    project: my-project
+    entrypoint: main.ts # the entrypoint must be relative to the root directory
+    root: path/to/your/directory
+```
 
-```yml
+### Filter content with `include` and `exclude`
+
+```yaml
 - name: Deploy to Deno Deploy
   uses: denoland/deployctl@v1
   with:
     project: my-project
-    entrypoint: https://deno.land/std/http/file_server.ts
-    root: dist
+    entrypoint: main.ts # the entrypoint must be relative to the root directory
+    include: |
+      main.ts
+      dist
+    exclude: |
+      **/node_modules
 ```
 
-If you want to use [import maps](https://github.com/WICG/import-maps):
+### Use external entrypoint
```

Comment:
It would be interesting to talk about the specific file_server use case, or just link to the tutorial (https://docs.deno.com/deploy/tutorials/static-site)